File: src/order-analytics-api/OrderAnalytics.Adapters/DAO/Tables/ProductTable.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Numerics;

namespace OrderAnalytics.Adapters.DAO.Tables;

[Table("tb_products")]
public class ProductTable
{
    [Key]
    [Column("id_product")]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int Id { get; set; }

    [Required]
    [Column("tx_name")]
    public string Name { get; set; } = string.Empty;
    
    [Required]
    [Column("tx_quantity")]
    public BigInteger Quantity { get; set; }
    
    [Required]
    [Column("tx_price")]
    public decimal Price { get; set; }
    
    [ForeignKey(nameof(OrderId))]
    [Column("fk_id_order")]
    public OrderTable Order { get;set; }
    public int OrderId { get; set; }
}

File: src/order-processor-service/src/main/java/com/gabrielaraujo/order_processor_service/adapters/dao/tables/ProductTable.java
package com.gabrielaraujo.order_processor_service.adapters.dao.tables;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.math.BigInteger;

@Entity
@Table(name = "tb_products")
@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ProductTable {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    @Column(name = "id_product")
    private int id;

    @Column(name = "tx_name", nullable = false)
    private String name;

    @Column(name = "tx_quantity", nullable = false)
    private BigInteger quantity;

    @Column(name = "tx_price", nullable = false)
    private BigDecimal price;
}
